src/components/products/SortButton.tsx

```diff
@@ -0,0 +1,74 @@
+import sortIcon from "../../assets/sort-icon.svg";
+import chevronDownIcon from "../../assets/chevron-down.svg";
+import { Dispatch, useState } from "react";
+import { Scooter } from "../../data/Scooters";
+import { Actions } from "../../reducers/productsReducer";
+
+export interface Sort {
+  method: (a: Scooter, b: Scooter) => number;
+  name: string;
+}
+
+interface SortButtonProps {
+  dispatch: Dispatch<Actions>;
+  products: Scooter[];
+}
+
+const sortFunctions = [
+  {
+    method: (a: Scooter, b: Scooter) => a.amount - b.amount,
+    name: "Best choice",
+  },
+  {
+    method: (a: Scooter, b: Scooter) => b.amount - a.amount,
+    name: "Popular",
+  },
+];
+
```

Comment:
Could you please add comments to enable the identification of sorting functions?